From e4f33a7e87d738fb2fb04a8401b6969d31872ffd Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 28 Feb 2023 18:27:22 +0100
Subject: [PATCH 0266/1141] arm64: dts: st: add rpmsg i2c bus for
 stm32mp257f-ev

Add rpmsg i2c bus support for communication with the Cortex-M33.
Linux drivers can register to this bus to simulate an I2C link over
rpmsg.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Ibd7bdc39526677817a9bb609e08c489deba17b9d
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/310435
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm64/boot/dts/st/stm32mp257f-ev.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp257f-ev.dts b/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
index ac481a699db4..0c391185f949 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev.dts
@@ -172,6 +172,15 @@ intc_rpmsg: interrupt-controller@0 {
 		#interrupt-cells = <1>;
 		interrupt-controller;
 	};
+
+	i2c_rpmsg: i2c@1 {
+		compatible = "rpmsg,i2c-controller";
+		reg = <1 0>;
+		rpmsg,dev-id = "rpmsg_i2c";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "okay";
+	};
 };
 
 &pcie_rc {
-- 
2.39.2

